test/helpers/kubectl.go

```diff
@@ -4328,8 +4328,17 @@ func validateCiliumSvc(cSvc models.Service, k8sSvcs []v1.Service, k8sEps []v1.En
 	}
 
 	for _, k8sSvc := range k8sSvcs {
-		if k8sSvc.Spec.ClusterIP == cSvc.Status.Realized.FrontendAddress.IP {
-			k8sService = &k8sSvc
+		clusterIPs := k8sSvc.Spec.ClusterIPs
+		if clusterIPs == nil || len(clusterIPs) == 0 {
+			clusterIPs = []string{k8sSvc.Spec.ClusterIP}
+		}
+		for _, clusterIP := range clusterIPs {
```

Comment:
Isn't this loop duplicated in line 4344? I like your new code better so I think it's safe to delete lines 4344-4353

Reply:
:exploding_head: good catch! Let me check why it didn't work on the v1.11 backport.